src/java.base/share/classes/java/util/concurrent/CancellationException.java

```diff
@@ -60,4 +60,16 @@ public CancellationException() {}
     public CancellationException(String message) {
         super(message);
     }
+
+    /**
+     * Only intended for in-package usage.
+     * Constructs a {@code CancellationException} with the specified detail
+     * message and CancellationException cause.
+     *
+     * @param message the detail message
+     * @param cause the underlying cancellation exception
+     */
+    CancellationException(String message, CancellationException cause) {
```

Comment:
Can we remove the message argument, which is always empty?

Reply:
Good question. So what I did was to initially omit it, but then the message becomes the type of the cause, which didn't look right. Then I passed in the empty-string in the constructor, but that seemed too restrictive, since the overload is for in-package use only, I opted for some flexibility in case we want to customize the message on `get` vs `join`.